Running total of score on review form

bickelj opened the pull request:
Live score label added to edit review form

Fixes #3963

## Test Steps
- As a Reviewer, add a review
- Change the values in the drop-downs see the live score updates on the right.
- As a Reviewer, edit a review
- Change the values again, see the live score updates again.

gmurtaza00 committed:
Added a label to keep running total of scores on review form

Issue #3963

Co-authored-by: Jesse Bickel <jbickel@opentechstrategies.com>
    Upstream lint and directory changes.

gmurtaza00 committed:
Score label added to edit review form and selectors filtering improved

Issue #3963

Co-authored-by: Jesse Bickel <jbickel@opentechstrategies.com>
    Upstream lint and directory changes.

gmurtaza00 committed:
length check of recommedation selector removed

Issue #3963

gmurtaza00 committed:
Condition text changed

Issue #3963

gmurtaza00 committed:
NaN values and ScoreWithoutTextBlock values handled

Issue #3963

frjo commented on hypha/static_src/javascript/review-form-score.js:
```diff
@@ -0,0 +1,30 @@
+        $('.form--score-box').text('Score: ' + score);
+    }
+
+})(jQuery);
```
Please rewrite in vanillajs or alpinejs. We are trying to face out jQuery.

bickelj replied:
I was afraid you'd say that...

frjo commented:
Run `pre-commit install` and you never will have a lint failure in a PR again :-).

frjo commented on hypha/static_src/sass/components/_form.scss:
```diff
@@ -38,13 +38,28 @@
+        padding: 0.5em 60px;
+        font-weight: 700;
+        position: sticky;
+        position: -webkit-sticky;
```
Not needed. Webkit supports `sticky` since some years.

bickelj replied:
Lint found the same once I ran it.

frjo commented on hypha/static_src/sass/components/_form.scss:
```diff
@@ -38,13 +38,28 @@
+        background-color: #1d79a8;
+        color: #fff;
+        padding: 0.5em 60px;
+        font-weight: 700;
```
`$weight--bold`

frjo commented on hypha/static_src/sass/components/_form.scss:
```diff
@@ -38,13 +38,28 @@
+        display: none;
+        background-color: #1d79a8;
+        color: #fff;
+        padding: 0.5em 60px;
```
Please use `rem` for all paddings.

frjo replied:
We also mostly use tailwind classes for new styling nowadays. But if a component already exists like here it is ok to build on that as well.

frjo commented on hypha/static_src/sass/components/_form.scss:
```diff
@@ -38,13 +38,28 @@
+        position: sticky;
+        position: -webkit-sticky;
+        top:0;
+        margin-top: 10px
```
`rem` and `margin-block-start`

bickelj commented:
@frjo I or someone will likely need to come back to this one.

bickelj commented:
c8b5694 included all 5 commits and fixed the linting, there are still several issues mentioned by @frjo above.

bickelj committed:
Use pure JavaScript

Refactored by dolphin-mixtral:8x7b cfada4ba31c7 on ollama 0.3.7-rc5.

Reformatted using prettier.

Issue #3963

bickelj committed:
Fix some CSS

Issue #3963

bickelj commented:
@frjo I think I got all the changes this time, what do you think?

frjo commented on hypha/static_src/sass/components/_wrapper.scss:
```diff
@@ -127,6 +127,13 @@
+    &--flex {
+        display: flex;
+        align-items: start;
+        column-gap: 25px;
```
A rem value is preferred. Just "gap" works as well unless rows should not have gaps.

frjo commented on hypha/static_src/javascript/review-form-score.js:
```diff
@@ -0,0 +1,38 @@
+
+        // remove recommendation select box from array
+        filtered_selectors = selectors.filter(
+            (selector) => selector[0].text !== "Need More Info"
```
What happens if this string is translated. Can we remove the element in a less fragile way?

bickelj replied:
I'm not even sure this code had any effect so I removed it.

frjo commented on hypha/apply/review/templates/review/review_edit_form.html:
```diff
@@ -42,5 +42,10 @@ <h2>{{ value }}</h2>
             {% endif %}
             <button class="button button--submit button--top-space button--primary" type="submit" name="submit">{% trans "Submit" %}</button>
         </form>
+        <div class="form--score-box">Score:0</div>
```
Space after colon.

frjo commented on hypha/apply/review/templates/review/review_form.html:
```diff
@@ -46,5 +46,9 @@ <h2>{{ value }}</h2>
         {% else %}
             <p>{% trans "You have already posted a review for this submission" %}</p>
         {% endif %}
+        <div class="form--score-box">Score:0</div>
```
Space after colon.

bickelj committed:
Use rem instead of px in CSS

bickelj committed:
Add a space

bickelj committed:
Remove selector filtering

This code did not seem to have any effect, at least vs vanilla Hypha.

frjo commented on hypha/static_src/javascript/review-form-score.js:
```diff
@@ -0,0 +1,31 @@
+
+        // Update the text in .form--score-box with the current score
+        document.querySelector(".form--score-box").textContent =
+            "Score: " + score;
```
Can we have the js only update the numeric value and not the text? Would we could to only add "Score" in the template so it can be translatable.

frjo replied:
An option to show total score or average score would be nice.

I think this also this feature should be off by default.

bickelj replied:
The text is now translatable. I added an on-page button to switch between total and average (default total), though the style of it leaves something to be desired. I also added a new boolean system setting `REVIEW_SHOW_RUNNING_TOTAL` for it to be enabled. I suppose an alternative implementation would be to let the system setting have three settings (`None`, `Total`, and `Average`) but I hereby commend improvements to be an exercise for the reader.

frjo commented on hypha/apply/review/templates/review/review_form.html:
```diff
@@ -46,5 +46,9 @@ <h2>{{ value }}</h2>
         {% else %}
             <p>{% trans "You have already posted a review for this submission" %}</p>
         {% endif %}
+        <div class="form--score-box">Score: 0</div>
```
Make "Score:" translatable.

bickelj committed:
Make review "Score:" translatable

Issue #3963

bickelj committed:
Add button to toggle between total and average

Allow any reviewer to switch view between total and average while on
the page.

Issue #3963

bickelj committed:
Add setting `REVIEW_SHOW_RUNNING_TOTAL`

When set to `True`, a running score appears on the review edit pages.
Default is `False`.

Issue #3963

frjo commented:
Move to https://github.com/HyphaApp/hypha/pull/4169
